client/data/newsletter-categories/use-mark-as-newsletter-category-mutation.tsx

```diff
@@ -28,8 +29,32 @@ const useMarkAsNewsletterCategoryMutation = ( siteId: string | number ) => {
 
 			return response;
 		},
-		onMutate: async () => {
+		onMutate: async ( id: number ) => {
```

Comment:
`id` can be confusing, maybe you could rename it to `categoryId` to make it more readable.